From zaitcev@redhat.com Tue Jun  6 18:18:46 2006
Date: Tue, 6 Jun 2006 18:18:33 -0700
From: Pete Zaitcev <zaitcev@redhat.com>
To: greg@kroah.com
Subject: USB: io_edgeport touch-up
Message-Id: <20060606181833.b2ce6c13.zaitcev@redhat.com>

Al Borchers suggested to do this in his review of my previous patch.
I guess that I skipped this initially because of my visceral dislike
of sizeof(data). But in this case it seems well localized.

Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/io_edgeport.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/io_edgeport.c
+++ gregkh-2.6/drivers/usb/serial/io_edgeport.c
@@ -2607,15 +2607,15 @@ static void get_manufacturing_desc (stru
 		dbg("  BoardRev:       %d", edge_serial->manuf_descriptor.BoardRev);
 		dbg("  NumPorts:       %d", edge_serial->manuf_descriptor.NumPorts);
 		dbg("  DescDate:       %d/%d/%d", edge_serial->manuf_descriptor.DescDate[0], edge_serial->manuf_descriptor.DescDate[1], edge_serial->manuf_descriptor.DescDate[2]+1900);
-		unicode_to_ascii(string, 30,
+		unicode_to_ascii(string, sizeof(string),
 		    edge_serial->manuf_descriptor.SerialNumber,
 		    edge_serial->manuf_descriptor.SerNumLength/2);
 		dbg("  SerialNumber: %s", string);
-		unicode_to_ascii(string, 30,
+		unicode_to_ascii(string, sizeof(string),
 		    edge_serial->manuf_descriptor.AssemblyNumber,
 		    edge_serial->manuf_descriptor.AssemblyNumLength/2);
 		dbg("  AssemblyNumber: %s", string);
-		unicode_to_ascii(string, 30,
+		unicode_to_ascii(string, sizeof(string),
 		    edge_serial->manuf_descriptor.OemAssyNumber,
 		    edge_serial->manuf_descriptor.OemAssyNumLength/2);
 		dbg("  OemAssyNumber:  %s", string);
